### DISCUSSION TITLE
[HELP] drag and drop

### DISCUSSION THREAD

**Discussion by jellybobbin** [gtk-rs/gtk4-rs#1480] [Q&A]
Are there any examples of drag and drop, such as dragging files to an area to achieve file upload.

**Comment by philippe06150** [comment]
Hi,

never used drag and drop, seems to me level 2 and I am still on level 1

have a look on GTK4 doc : https://docs.gtk.org/gtk4/drag-and-drop.html
I think it can be an input where to look for in gtk-rs doc

it seems to start there : https://gtk-rs.org/gtk4-rs/git/docs/gtk4/struct.DragSource.html
examples are in C, need some work and some knowledge to translate in RUST :-(

Strength and honor.
